linkerd/core/src/main/scala/io/buoyant/linkerd/RequestActiveTracer.scala

```diff
@@ -0,0 +1,265 @@
+package io.buoyant.linkerd
+
+import com.twitter.finagle._
+import com.twitter.finagle.client.Transporter.EndpointAddr
+import com.twitter.finagle.http._
+import com.twitter.finagle.http.Fields.MaxForwards
+import com.twitter.finagle.http.Status
+import com.twitter.finagle.http.util.StringUtil
+import com.twitter.finagle.naming.buoyant.DstBindingFactory
+import com.twitter.util._
+import io.buoyant.namer.DelegateTree._
+import io.buoyant.namer.{DelegateTree, Delegator}
+import io.buoyant.router.RouterLabel
+import io.buoyant.router.RoutingFactory
+import io.buoyant.router.RoutingFactory.BaseDtab
+import io.buoyant.router.context.{DstBoundCtx, DstPathCtx}
+
+/**
+ * Intercepts Http TRACE requests with a Max-Forwards and l5d-add-context header.
```

Comment:
I would expect the header be named `l5d-max-forwards` to be consistent with other linkerd-specific headers. Is the `max-forwards` header standardized more widely?

Reply:
Yes, from https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html

>The final recipient is either the origin server or the first proxy or gateway to receive a Max-Forwards value of zero (0) in the request (see section 14.31).

Reply:
Yea, `Max-Forwards is the default header associated with the `TRACE` method according to the spec. The header is what proxies use to figure out if they should forward a trace request or just reflect it back to the client.

Reply:
> The Max-Forwards request-header field provides a mechanism with the TRACE (section 9.8) and OPTIONS (section 9.2) methods to limit the number of proxies or gateways that can forward the request to the next inbound server. This can be useful when the client is attempting to trace a request chain which appears to be failing or looping in mid-chain.

Reply:
Excellent! thank you. it might be helpful to include that RFC link at the bottom of the docstring.